internal/pr/promote/interactive_prompt_provider.go

```diff
@@ -0,0 +1,82 @@
+package promote
+
+import (
+	"fmt"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/nestoca/joy/internal/style"
+)
+
+type InteractivePromptProvider struct {
+}
+
+func (s *InteractivePromptProvider) WhetherToCreateMissingPullRequest() (bool, error) {
+	prompt := &survey.Confirm{
+		Message: "No pull request found for current branch, create one?",
+		Default: true,
+	}
+	var shouldCreate bool
+	err := survey.AskOne(prompt, &shouldCreate)
+	if err != nil {
+		return false, fmt.Errorf("prompting user to create pull request: %w", err)
+	}
+	return shouldCreate, nil
+}
+
+func (s *InteractivePromptProvider) WhichEnvironmentToPromoteTo(environments []string, preSelectedEnv string) (string, error) {
+	none := "[none]"
+	if preSelectedEnv == "" {
+		preSelectedEnv = none
+	}
+	options := append([]string{none}, environments...)
+	prompt := &survey.Select{
+		Message: "Select environment to auto-promote builds of pull request to:",
+		Options: options,
+		Default: preSelectedEnv,
+	}
+	var selectedEnv string
+	err := survey.AskOne(prompt, &selectedEnv)
+	if err != nil {
+		return "", fmt.Errorf("prompting user to select environment: %w", err)
+	}
+	if selectedEnv == none {
+		selectedEnv = ""
+	}
+	return selectedEnv, nil
+}
+
+func (s *InteractivePromptProvider) ConfirmDisablingPromotionOnOtherPullRequest(branch, env string) (bool, error) {
+	prompt := &survey.Confirm{
+		Message: fmt.Sprintf("Another pull request for branch %s is already auto-promoting to %s environment, disable it?", style.Resource(branch), style.Resource(env)),
+		Default: false,
+	}
+	var shouldDisable bool
+	err := survey.AskOne(prompt, &shouldDisable)
+	if err != nil {
+		return false, fmt.Errorf("prompting user to confirm disabling promotion on other pull request: %w", err)
+	}
+	return shouldDisable, nil
```

Comment:
In which case would a user say no and still want to continue?

I would assume this prompt would be a confirmation, as in Yes (continue) or No (abort).